server/isomorphic-routing/index.js

```diff
@@ -37,6 +39,15 @@ export function serverRouter( expressApp, setUpRoute, section ) {
 	};
 }
 
+function setRouteMiddleware( context, next ) {
+	context.store.dispatch( setRouteAction(
+		context.pathname,
+		context.query
+	) );
+
+	next();
```

Comment:
That's not what I meant...
if we're not `return`ing next's result, we're denying next's middleware chance to return a value, consider all the examples at `redux` docs:
http://redux.js.org/docs/advanced/Middleware.html#seven-examples